src/app/admin/services/controller/equipment-controller.service.ts

```diff
@@ -0,0 +1,225 @@
+import { Injectable } from '@angular/core';
+import { ConfirmationModalComponent } from '@shared/components/confirmation-modal/confirmation-modal.component';
+import { MatDialog } from '@angular/material/dialog';
+import { ArchiveEquipmentModalComponent } from '@app/admin/components/archive-equipment-modal/archive-equipment-modal.component';
+import { EquipmentModal } from '@app/admin/constants/equipment-modal.enum';
+import { AdminApi } from '@app/admin/services';
+import { BehaviorSubject, Observable, filter, of, map, switchMap, tap } from 'rxjs';
+import { Equipment } from '@app/catalog/models/equipment';
+import { BlockEquipmentModalComponent } from '@app/admin/components/block-equipment-modal/block-equipment-modal.component';
+import { Dictionary } from '@app/shared/types';
+import { TableAction } from '@shared/models/table-action';
+import { EquipmentAction } from '@shared/constants/action.enum';
+import { NotificationsService } from '@app/shared/services/notifications/notifications.service';
+import { UntilDestroy, untilDestroyed } from '@app/shared/until-destroy/until-destroy';
+import { OrderNotificationModalComponent } from '@app/admin/components/order-notification-modal/order-notification-modal.component';
+import { TableRow } from '@app/shared/models/table-row';
+import { EquipmentStatusId } from '@app/shared/models/equipment-status-ids';
+import { equipmentStatusIdDefaultValue } from '@app/shared/constants/equipment-status-id';
+import { EquipmentStatus } from '@app/admin/types/equipment-status';
+import { Period } from '@app/shared/models/period';
+import { UnavailableDates } from '@app/features/date-range/models';
+
+@UntilDestroy
+@Injectable()
+export class EquipmentController {
+  private equipmentDataSubj$ = new BehaviorSubject<TableRow[]>([]);
+  categoryDictionary: Dictionary<string> = {};
```

Comment:
Do we really need this Dictionary type? In all current usage cases it's like a simple object.

Reply:
yes we need. as I can see we can add new category and correct existing ones